org.opentosca.bus.management.api.osgievent/src/org/opentosca/bus/management/api/osgievent/MBEventHandler.java

```diff
@@ -157,8 +222,518 @@ public void handleEvent(final Event event) {
+
+
+
+            String correlationID = String.valueOf(System.currentTimeMillis());
+            QName planId = null;
+            PlanType planType = null;
+            Map<String, String> inputs = null;
+
+            if (endpoint != null) {
+                planId = endpoint.getPlanId();
+                planType = PlanType.fromString(endpoint.getMetadata().get("PLANTYPE"));
+
+                inputs = new HashMap<String, String>();
+
+                for (String input : this.toStringCollection(endpoint.getMetadata().get("INPUTS"), ",")) {
+                    inputs.put(input, null);
+                }
+
+            } else {
+                try {
+                    BPELPlan adaptationPlan =
+                        (BPELPlan) importer.generateAdaptationPlan(instance.getCsarId(), instance.getTemplateId(),
+                                                                   currentConfigNodeIds, currentConfigRelationIds,
+                                                                   targetConfigNodeIds, targetConfigRelationIds);
+
+                    planType = PlanType.fromString(adaptationPlan.getType().getString());
+                    inputs = this.createInput(adaptationPlan);
+                    Path tempFile = Files.createTempFile(adaptationPlan.getId(), ".zip");
+                    exporter.exportToPlanFile(tempFile.toUri(), adaptationPlan);
+                    BpelPlanEnginePlugin deployPlugin = this.getBpelDeployPlugin();
+
+                    Map<String, String> endpointMetadata =
+                        this.toEndpointMetadata(currentConfigNodeIds, currentConfigRelationIds, targetConfigNodeIds,
+                                                targetConfigRelationIds);
+
+                    endpointMetadata.put("PLANTYPE", planType.toString());
+                    endpointMetadata.put("INPUTS", this.toCSV(inputs.keySet()));
+
+                    deployPlugin.deployPlanFile(tempFile, instance.getCsarId(), endpointMetadata);
+
+                    planId = new QName(tempFile.getFileName().toString());
+
+
+                }
+                catch (SystemException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+                catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+                catch (JAXBException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+            }
+
+            Map<String, String> requestBody = null;
+            try {
+                requestBody = this.createRequestBody(instance.getCsarId(), instance.getTemplateId(), instance.getId(),
+                                                     inputs, correlationID);
+            }
+            catch (UnsupportedEncodingException e1) {
```

Comment:
I'd expect that to be handled within `createRequestBody`

Reply:
Moved exception handling to private method